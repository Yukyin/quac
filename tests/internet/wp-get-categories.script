#!/bin/bash

# Test that wp-get-categories script and underlying functionality works. We
# assume that the categories of the English main page are constant.
#
# Copyright © Los Alamos National Security, LLC, and others.

. ./environment.sh

set +e


# Basic category set
x wp-get-categories en+Main_page
x wp-get-categories --hidden en+Main_page

# Multi-page queries
x wp-get-categories --hidden -p2 en+Main_page  # fills all pages
x wp-get-categories --hidden -p5 en+Main_page  # last page not full

# Not in English
x wp-get-categories ro+Pagina_principală

# Various encodings of article
x wp-get-categories "'ro+Pagina principală'"
x wp-get-categories ro+Pagina_principal%C4%83
x wp-get-categories ro+Pagina%20principal%C4%83

# No categories
x wp-get-categories pt+Página_principal

# Non-existent article
x wp-get-categories en+NOTFOUND

# Errors
x wp-get-categories
x wp-get-categories Main_page
x wp-get-categories -p0 en+Main_page
x wp-get-categories -p foo en+Main_page

exit 0
