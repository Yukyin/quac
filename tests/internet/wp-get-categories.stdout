$ wp-get-categories en+Main_page
en+Category:Main Page
$ wp-get-categories --hidden en+Main_page
en+Category:Main Page
en+Category:Protected redirects
en+Category:Redirects from other capitalisations
en+Category:Unprintworthy redirects
en+Category:Wikipedia indefinitely move-protected pages
en+Category:Wikipedia protected pages
$ wp-get-categories --hidden -p2 en+Main_page
en+Category:Main Page
en+Category:Protected redirects
en+Category:Redirects from other capitalisations
en+Category:Unprintworthy redirects
en+Category:Wikipedia indefinitely move-protected pages
en+Category:Wikipedia protected pages
$ wp-get-categories --hidden -p5 en+Main_page
en+Category:Main Page
en+Category:Protected redirects
en+Category:Redirects from other capitalisations
en+Category:Unprintworthy redirects
en+Category:Wikipedia indefinitely move-protected pages
en+Category:Wikipedia protected pages
$ wp-get-categories ro+Pagina_principală
ro+Categorie:Pagina principală
$ wp-get-categories 'ro+Pagina principală'
ro+Categorie:Pagina principală
$ wp-get-categories ro+Pagina_principal%C4%83
ro+Categorie:Pagina principală
$ wp-get-categories ro+Pagina%20principal%C4%83
ro+Categorie:Pagina principală
$ wp-get-categories pt+Página_principal
$ wp-get-categories en+NOTFOUND
wpcat FATAL    article not found
$ wp-get-categories
usage: wp-get-categories [-d] [-p N] [-h] [--config FILE] [--notimes]
                         [--unittest] [--verbose]
                         URL
wp-get-categories: error: the following arguments are required: URL
$ wp-get-categories Main_page
wpcat FATAL    no language specified
$ wp-get-categories -p0 en+Main_page
wpcat FATAL    invalid page size
$ wp-get-categories -p foo en+Main_page
usage: wp-get-categories [-d] [-p N] [-h] [--config FILE] [--notimes]
                         [--unittest] [--verbose]
                         URL
wp-get-categories: error: argument -p/--page-size: invalid int value: 'foo'
